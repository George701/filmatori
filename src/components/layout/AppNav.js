import React from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    Container
} from 'reactstrap';

const AppNav = () => {
    return (
        <Navbar color="dark" dark expand="sm" className="mb-5">
            <Container>
                <NavbarBrand href="/">
                    <h4 className="brand-color">FILMATORI</h4>
                </NavbarBrand>
                <NavbarToggler />
                <Collapse navbar>
                    <Nav className="ml-auto" navbar>
                        <NavItem>
                            <NavLink href='/hall'>
                                Hall
                            </NavLink>
                        </NavItem>
                        <NavItem>
                            <NavLink href='/op-2'>
                                Opt 2
                            </NavLink>
                        </NavItem>
                    </Nav>
                </Collapse>
            </Container>
        </Navbar>
    )
}


export default AppNav;
