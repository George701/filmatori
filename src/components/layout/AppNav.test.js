import React from 'react';
import { shallow } from '../../enzyme';
import AppNav from './AppNav';

describe('AppNav', () => {
    const appNav = shallow(<AppNav />);

    it('renders properly', () => {
        expect(appNav).toMatchSnapshot();
    });

    it('displays NavbarBrand Reactstrap component', () => {
        expect(appNav.find('NavbarBrand').exists()).toBe(true);
    });

    it('displays NavbarBrand Reactstrap component, which refers to route `/`', () => {
        expect(appNav.find('NavbarBrand').props().href).toEqual('/');
    });

    it('displays two NavItem Reactstrap components', () => {
        expect(appNav.find('NavItem').length).toEqual(2);
    });

    it('displays first NavLink Reactstrap component, which refers to route `/hall`', () => {
        expect(appNav.find('NavLink').at(0).props().href).toEqual('/hall');
    });

    it('displays second NavLink Reactstrap component, which refers to github repository', () => {
        const link = 'https://github.com/George701/filmatori';
        expect(appNav.find('NavLink').at(1).props().href).toEqual(link);
    });
});