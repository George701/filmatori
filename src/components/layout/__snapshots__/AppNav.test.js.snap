// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav renders properly 1`] = `ShallowWrapper {}`;
